cmd/internal/applicationsnapshot/source.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright © 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applicationsnapshot
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/hashicorp/go-getter"
+)
+
+const SCHEMA = "https://raw.githubusercontent.com/redhat-appstudio/release-service/main/config/crd/bases/appstudio.redhat.com_applicationsnapshots.yaml"
+
+type fetcherFn func() (*os.File, error)
+
+type schemaSource struct {
+	fetch fetcherFn
+}
+
+// NewSchemaSource constructs the default schemaSource
+func NewSchemaSource() schemaSource {
+	return schemaSource{
+		fetch: fetch,
+	}
+}
+
+// fetch the latest CRD and return the file
+//go:generate build
+func fetch() (*os.File, error) {
+	// Create a tempfile for the CRD
+	file, err := ioutil.TempFile(os.TempDir(), "ec-cli-applicationsnapshot")
+	if err != nil {
+		return nil, err
+	}
+
+	// Download the CRD
+	client := getter.Client{
+		Ctx:      context.Background(),
+		Src:      SCHEMA,
+		Dst:      file.Name(),
+		Insecure: false,
+		Getters: map[string]getter.Getter{
+			"https": &getter.HttpGetter{},
+		},
+	}
```

Comment:
Simpler version would be:

```go
	getter.Get(file.Name(), SCHEMA, getter.WithContext(ctx))
```

`ctx` can be obtained from the Command instance (`cmd.Context()`)

Reply:
This doesn't seem to work. getting ` no source URL was returned`